pkg/controller/framework/controller_test.go

```diff
@@ -0,0 +1,115 @@
+/*
+Copyright 2015 Google Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package framework_test
+
+import (
+	"fmt"
+	"sync"
+	"time"
+	// "testing"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/client/cache"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/controller/framework"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/runtime"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/util"
+)
+
+func Example() {
+	// source simulates an apiserver object endpoint.
+	source := framework.NewFakeControllerSource()
+
+	// This will hold the downstream state, as we know it.
+	downstream := cache.NewStore(cache.MetaNamespaceKeyFunc)
+
+	// This will hold incoming changes. Note how we pass downstream in as a
+	// KeyLister, that way resync operations will result in the correct set
+	// of update/delete deltas.
+	fifo := cache.NewDeltaFIFO(cache.MetaNamespaceKeyFunc, nil, downstream)
+
+	// Let's do threadsafe output to get predictable test results.
+	outputSetLock := sync.Mutex{}
+	outputSet := util.StringSet{}
+
+	cfg := &framework.Config{
+		Queue:            fifo,
+		ListerWatcher:    source,
+		ObjectType:       &api.Pod{},
+		FullResyncPeriod: time.Millisecond * 100,
+		RetryOnError:     false,
+
+		// Let's implement a simple controller that just deletes
+		// everything that comes in.
+		Process: func(obj interface{}) error {
```

Comment:
So if I want to use this in kube2sky, for example, do I literally just rip off most of this function?  That still feels like a lot of boilerplate - easy to get wrong.

Can it be reduced to "call one function, passing a set of function pointers (or an interface), and have my functions called on add/delete/update" ?